pagerduty/resource_pagerduty_event_orchestration_integration.go

```diff
@@ -0,0 +1,311 @@
+		ReadContext:   resourcePagerDutyEventOrchestrationIntegrationRead,
+		UpdateContext: resourcePagerDutyEventOrchestrationIntegrationUpdate,
+		DeleteContext: resourcePagerDutyEventOrchestrationIntegrationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourcePagerDutyEventOrchestrationIntegrationImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"event_orchestration": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: addIntegrationMigrationWarning(),
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"label": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parameters": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"routing_key": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func addIntegrationMigrationWarning() schema.SchemaValidateDiagFunc {
+	return func(v interface{}, p cty.Path) diag.Diagnostics {
+		var diags diag.Diagnostics
+
+		diags = append(diags, diag.Diagnostic{
+			Severity:      diag.Warning,
+			Summary:       "Modifying the event_orchestration property of the 'pagerduty_event_orchestration_integration' resource will cause all future events sent with this integration's routing key to be evaluated against the new Event Orchestration.",
+			AttributePath: p,
+		})
+
+		return diags
+	}
+}
+
+func getEventOrchestrationIntegrationPayloadData(d *schema.ResourceData) (string, *pagerduty.EventOrchestrationIntegration) {
+	orchestrationId := d.Get("event_orchestration").(string)
+
+	integration := &pagerduty.EventOrchestrationIntegration{
+		Label: d.Get("label").(string),
+	}
+
+	return orchestrationId, integration
+}
+
+func fetchPagerDutyEventOrchestrationIntegration(ctx context.Context, d *schema.ResourceData, meta interface{}, oid string, id string, compareLabels bool) (*pagerduty.EventOrchestrationIntegration, error) {
+	client, err := meta.(*Config).Client()
+	if err != nil {
+		return nil, err
+	}
+
+	if integration, _, err := client.EventOrchestrationIntegrations.GetContext(ctx, oid, id); err != nil {
+		return nil, err
+	} else if integration != nil {
+		lbl := d.Get("label").(string)
+		if compareLabels && strings.Compare(integration.Label, lbl) != 0 {
+			return integration, fmt.Errorf("Integration '%s' for PagerDuty Event Orchestration '%s' error - stored label '%s' doesn't match resource label '%s'.", id, oid, integration.Label, lbl)
+		} else {
+			d.SetId(id)
+			d.Set("event_orchestration", oid)
+			setEventOrchestrationIntegrationProps(d, integration)
+			return integration, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Reading Integration '%s' for PagerDuty Event Orchestration '%s' returned `nil`.", id, oid)
+}
```

Comment:
This function doesn't have retries in it because it's called from inside `.RetryContext()` methods that implement retries. Also specifically in case with integrations if we can't read the created/updated integration the first time it means that the integration wasn't created/updated so reading it repeatedly won't make any difference. Instead we want to fail fast and retry creating/updating an integration.